### ISSUE TITLE
Poor updating/scrolling/rendering of notebooks

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#148164]
* open a notebook on insiders.vscode.dev
* run a few cells
* pull latest changes - the notebook refreshed because of [vscode-remote-repositories-github/issues/80](https://github.com/microsoft/vscode-remote-repositories-github/issues/80)
* scroll up

During the update/refresh and while scrolling, the notebook looks and feels bad. Markdown headers jump up/down and overlay. This happens everywhere: safari, chrome, edge, and desktop and isn't some random fluke but repros 

Recording:

https://user-images.githubusercontent.com/1794099/165329792-42ddd5fb-51ce-4041-b707-5ce2c10194d0.mov

Selection of captures:
<img width="526" alt="Screenshot 2022-04-26 at 17 03 51" src="https://user-images.githubusercontent.com/1794099/165331542-b7023f71-3cdd-49a8-b3f9-818a61fd6a43.png">
<img width="422" alt="Screenshot 2022-04-26 at 17 03 33" src="https://user-images.githubusercontent.com/1794099/165331561-87bc6461-cf73-4e0f-b01c-2d8561cb9f16.png">
<img width="309" alt="Screenshot 2022-04-26 at 17 03 43" src="https://user-images.githubusercontent.com/1794099/165331565-4b6489f5-a5b2-41da-9fcb-ad317f30545b.png">

**Comment by byehack** [comment]
True. it sometimes becomes very annoying with [cpu/memory leak.](https://github.com/microsoft/vscode/issues/147846)

**Comment by byehack** [comment]
https://user-images.githubusercontent.com/26390254/166409428-756c02bf-8b20-49bd-b6ae-83370a110041.mp4


[Untitled-1.zip](https://github.com/microsoft/vscode/files/8608414/Untitled-1.zip)

**Comment by roblourens** [comment]
We fixed a few things here, and I'll call this issue fixed. To trigger the same case as in the original report:

- Open some notebook, scroll to the bottom
- In a terminal, `touch <path to notebook>`
- The notebook may flicker a little as its model is reloaded
- Scroll up
- You should not see wild flickering of markdown cells

**Comment by jrieken** [comment]
This is much better and it's great to see progress. I wouldn't call the overall experience great - there is still the usual flicker when scrolling for the first time

**Comment by jrieken** [comment]
I don't know why it re-opened... Creating a separate issue for the "other" flicker
